### PR TITLE
chore: fix nav import and tests

### PR DESCRIPTION
**Pull request by jleach**
# Summary of Changes

The react docs note that we should be importing from `@react-navigation/native` so I've updated the imports, tests, and mock object accordingly. You can see how they use it in [the docs](https://reactnavigation.org/docs/use-navigation/).

# Related Issues

n/a

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [x] Updated documentation as needed for changed code and new or modified features;
- [x] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: packages/legacy/core/App/hooks/chat-messages.tsx [hunk 1]
```diff
@@ -90,9 +90,9 @@ export const useChatMessagesByConnection = (connection: ConnectionRecord): Exten
     const transformedMessages: Array<ExtendedChatMessage> = basicMessages.map((record: BasicMessageRecord) => {
       const role = getMessageEventRole(record)
       // eslint-disable-next-line
-      const linkRegex = /(?:https?\:\/\/\w+(?:\.\w+)+\S*)|(?:[\w\d\.\_\-]+@\w+(?:\.\w+)+)/gmi
+      const linkRegex = /(?:https?\:\/\/\w+(?:\.\w+)+\S*)|(?:[\w\d\.\_\-]+@\w+(?:\.\w+)+)/gim
```
Comment (bryce-mcmath) [inline packages/legacy/core/App/hooks/chat-messages.tsx:93]: Why this change?

Comment (jleach) [reply packages/legacy/core/App/hooks/chat-messages.tsx:93]: Good catch, I'm not sure what made this change. I'll revert it.

Comment (jleach) [reply packages/legacy/core/App/hooks/chat-messages.tsx:93]: I'm not sure what made this change. Maybe prettier putting them in alphabetical order? In any case, in JS the order of these options does not matter so I'll leave.

Comment (jleach) [reply packages/legacy/core/App/hooks/chat-messages.tsx:93]: Confirmed. These were put in alphabetical order on save in VSCode.

**Review by bryce-mcmath — APPROVED** [review]
